src/group_theory/commuting_probability.lean

```diff
@@ -72,3 +75,33 @@ begin
   rw [comm_prob, card_comm_eq_card_conj_classes_mul_card, nat.cast_mul, sq],
   exact mul_div_mul_right (card (conj_classes G)) (card G) (nat.cast_ne_zero.mpr card_ne_zero),
 end
+
+variables {G} (H : subgroup G)
+
+lemma subgroup.comm_prob_subgroup_le : comm_prob H ≤ comm_prob G * H.index ^ 2 :=
+begin
+  rw [comm_prob, comm_prob, div_le_iff, mul_assoc, ←mul_pow, ←nat.cast_mul, H.index_mul_card,
```

Comment:
can you add comments to the proofs (here and below) explaining what is going on in the proof (a quick informal sketch is enough)? A chain of rewrites is not always easy to follow...

Reply:
I've added some comments. Are they the sort of comment that you were thinking of?